File: src/main/java/ting/leetcode/java/arrays101/RotateImage.java
package ting.leetcode.java.arrays101;

public class RotateImage {
    public void rotate(int[][] matrix) {
        int matrixLength = matrix.length;
        for (int i = 0; i < matrixLength; i++) {
            for (int j = i; j < matrixLength; j++) {
                int temp = matrix[i][j];
                matrix[i][j] = matrix[j][i];
                matrix[j][i] = temp;
            }
        }

        for (int[] row : matrix) {
            int lastIndex = row.length - 1;
            for (int i = 0; i < row.length / 2; i++) {
                int temp = row[i];
                row[i] = row[lastIndex - i];
                row[lastIndex - i] = temp;
            }
        }
    }
}

File: src/main/kotlin/ting/leetcode/arrays101/RotateImage.kt
package ting.leetcode.arrays101

class RotateImage {
    fun rotate(matrix: Array<IntArray>): Unit {
        for (i in matrix.indices) {
            for (j in i until matrix.size) {
                matrix[i][j] = matrix[j][i]
                    .also { matrix[j][i] = matrix[i][j] }
            }
        }

        for (row in matrix) {
            row.reverse()
        }
    }
}
